Apps/WebClient/src/NewClientApp/src/components/site/CommunicationComponent.vue

```diff
@@ -0,0 +1,92 @@
+<script setup lang="ts">
+import { computed, ref, watch } from "vue";
+import { useRoute } from "vue-router";
+
+import { container } from "@/ioc/container";
+import { SERVICE_IDENTIFIER } from "@/ioc/identifier";
+import Communication, { CommunicationType } from "@/models/communication";
+import { ResultError } from "@/models/errors";
+import RequestResult from "@/models/requestResult";
+import { ICommunicationService, ILogger } from "@/services/interfaces";
+import { useAuthStore } from "@/stores/auth";
+import { useConfigStore } from "@/stores/config";
+import { useUserStore } from "@/stores/user";
+
+const logger = container.get<ILogger>(SERVICE_IDENTIFIER.Logger);
+const communicationService = container.get<ICommunicationService>(
+    SERVICE_IDENTIFIER.CommunicationService
+);
+const route = useRoute();
+const authStore = useAuthStore();
+const configStore = useConfigStore();
+const userStore = useUserStore();
+
+const bannerCommunication = ref<Communication | null>(null);
+const inAppCommunication = ref<Communication | null>(null);
+
+const displayInAppBanner = computed(() => {
+    return (
+        authStore.oidcIsAuthenticated &&
+        userStore.userIsRegistered &&
+        userStore.isValidIdentityProvider &&
+        !configStore.isOffline
+    );
+});
+const hasCommunication = computed(() =>
+    displayInAppBanner.value
+        ? inAppCommunication.value != null
+        : bannerCommunication.value != null
+);
+const text = computed(() =>
+    displayInAppBanner.value
+        ? inAppCommunication.value?.text
+        : bannerCommunication.value?.text
+);
+
+function setCommunication(
+    type: CommunicationType,
+    communication: Communication | null
+) {
+    if (type === CommunicationType.Banner) {
+        bannerCommunication.value = communication;
+    } else {
+        inAppCommunication.value = communication;
+    }
+}
+
+function fetchCommunication(type: CommunicationType): void {
+    communicationService
+        .getActive(type)
+        .then((requestResult: RequestResult<Communication | null>) =>
+            setCommunication(type, requestResult.resourcePayload)
+        )
+        .catch((err: ResultError) => {
+            if (err.statusCode === 429) {
```

Comment:
Considering the frequency of checking for 429 status code errors, I would argue this is a good bit of logic to embed into the httpDelegate.

* Future Improvement

Reply:
The issue is that different places need to handle 429s differently, but we could take another look at it.  We have some common handling of 429s in the stores.